Src/WindowsServer/WindowsServer.Shared.Tests/AzureInstanceMetadataTests.cs

```diff
@@ -0,0 +1,53 @@
+namespace Microsoft.ApplicationInsights.WindowsServer
+{
+    using Microsoft.ApplicationInsights.WindowsServer.Implementation;
+    using Microsoft.ApplicationInsights.WindowsServer.Mock;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+    using Assert = Xunit.Assert;
+
+    [TestClass]
+    public class AzureInstanceMetadataTests
+    {
+        [TestMethod]
+        public void GetAzureInstanceMetadataFieldsAsExpected()
+        {
+            HeartbeatProviderMock hbeatMock = new HeartbeatProviderMock();
+            AzureInstanceMetadataRequestMock azureInstanceRequestorMock = new AzureInstanceMetadataRequestMock();
+            AzureHeartbeatProperties azureIMSFields = new AzureHeartbeatProperties(azureInstanceRequestorMock, true);
+            int counter = 1;
+            foreach (string field in azureIMSFields.DefaultFields)
+            {
+                azureInstanceRequestorMock.ComputeFields.Add(field, $"testValue{counter++}");
+            }
+
+            var taskWaiter = azureIMSFields.SetDefaultPayload(new string[] { }, hbeatMock).ConfigureAwait(false);
+            Assert.True(taskWaiter.GetAwaiter().GetResult()); // no await for tests
+
+            foreach (string fieldName in azureIMSFields.DefaultFields)
+            {
+                Assert.True(hbeatMock.HbeatProps.ContainsKey(fieldName));
+                Assert.False(string.IsNullOrEmpty(hbeatMock.HbeatProps[fieldName]));
+            }
+        }
+
+        [TestMethod]
+        public void FailToObtainAzureInstanceMetadataFieldsAltogether()
```

Comment:
Will it be closer to the actual behavior if you make IMS request to fail via mock and then somehow prompt it to execute and validate that only default properties are present?

Reply:
Agreed. Throwing an exception in the Mock requestor approximates the behavior of a proper failure now.